kmath-tensors/src/commonMain/kotlin/space/kscience/kmath/tensors/api/TensorAlgebra.kt

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.tensors.api
+
+import space.kscience.kmath.operations.Algebra
+
+/**
+ * Algebra over a ring on [Tensor].
+ * For more information: https://proofwiki.org/wiki/Definition:Algebra_over_Ring
+ *
+ * @param T the type of items in the tensors.
+ */
+public interface TensorAlgebra<T>: Algebra<Tensor<T>> {
+
```

Comment:
I am against this in the interface because I just want to specify the functionality, and not the tensor creation API (it's very different from one framework to the other). I can add this to our implementation though in `DoubleTensorAlgebra`.